packages/builder/src/package.ts

```diff
@@ -76,19 +76,35 @@ export class PackageReference {
 
     if (!match || !match.groups?.name) {
       throw new Error(
-        `Invalid package name "${ref}". Should be of the format <package-name>:<version> or <package-name>:<version>@<preset>`
+        `Invalid package reference "${ref}". Should be of the format <package-name>:<version> or <package-name>:<version>@<preset>`
       );
     }
 
     const res: PartialRefValues = { name: match.groups.name };
 
+    const nameSize = res.name.length;
+    if (!(nameSize <= 32)) {
+      throw new Error(`Package reference "${ref}" is too long. Package name exceeds 32 bytes`);
+    }
+
     if (match.groups.version) res.version = match.groups.version;
+
+    const versionSize = res.name.length;
+    if (!(versionSize <= 32)) {
+      throw new Error(`Package reference "${ref}" is too long. Package version exceeds 32 bytes`);
+    }
+
     if (match.groups.preset) res.preset = match.groups.preset;
 
     return res;
   }
 
   static isValid(ref: string) {
+    try {
+      PackageReference.parse(ref);
+    } catch (err) {
+      return false;
+    }
     return !!PackageReference.PACKAGE_REGEX.test(ref);
```

Comment:
The regex is already being used inside the `.parse`:
```suggestion
    try {
      PackageReference.parse(ref);
      return true;
    } catch (err) {
      return false;
    }
```